parse/ValueRefPythonParser.cpp

```diff
@@ -66,3 +73,88 @@ condition_wrapper operator==(const value_ref_wrapper<int>& lhs, const value_ref_
+            }
+            return boost::python::object(value_ref_wrapper<double>(std::make_shared<ValueRef::Operation<double>>(O, std::move(operands))));
+        } else {
+            ErrorLogger() << "Unsupported type for min/max/oneof : " << boost::python::extract<std::string>(boost::python::str(args[0]))();
+
+            throw std::runtime_error(std::string("Not implemented ") + __func__);
+        }
+
+        return boost::python::object();
+    }
+
+    boost::python::object insert_game_rule_(const PythonParser& parser, const boost::python::tuple& args, const boost::python::dict& kw) {
+        auto name = boost::python::extract<std::string>(kw["name"])();
+        auto type_ = kw["type"];
+
+        if (type_ == parser.type_int) {
+            return boost::python::object(value_ref_wrapper<int>(std::make_shared<ValueRef::ComplexVariable<int>>(
+                "GameRule",
+                nullptr,
+                nullptr,
+                nullptr,
+                std::make_unique<ValueRef::Constant<std::string>>(name),
+                nullptr)));
+        } else if (type_ == parser.type_float) {
+            return boost::python::object(value_ref_wrapper<double>(std::make_shared<ValueRef::ComplexVariable<double>>(
+                "GameRule",
+                nullptr,
+                nullptr,
+                nullptr,
+                std::make_unique<ValueRef::Constant<std::string>>(name),
+                nullptr)));
+        } else {
+            ErrorLogger() << "Unsupported type for rule " << name << ": " << boost::python::extract<std::string>(boost::python::str(type_))();
+
+            throw std::runtime_error(std::string("Not implemented ") + __func__);
+        }
+
+        return boost::python::object();
+    }
+}
+
+void RegisterGlobalsValueRefs(boost::python::dict& globals, const PythonParser& parser) {
+    globals["NamedRealLookup"] = boost::python::raw_function(insert_named_lookup_<double>);
+    globals["Value"] = value_ref_wrapper<double>(std::make_shared<ValueRef::Variable<double>>(ValueRef::ReferenceType::EFFECT_TARGET_VALUE_REFERENCE));
+    globals["CurrentTurn"] = value_ref_wrapper<int>(std::make_shared<ValueRef::Variable<int>>(ValueRef::ReferenceType::NON_OBJECT_REFERENCE, "CurrentTurn"));
+
+    std::function<boost::python::object(const boost::python::tuple&, const boost::python::dict&)> f_insert_game_rule = [&parser](const boost::python::tuple& args, const boost::python::dict& kw) { return insert_game_rule_(parser, args, kw); };
+    globals["GameRule"] = boost::python::raw_function(f_insert_game_rule);
+    std::function<boost::python::object(const boost::python::tuple&, const boost::python::dict&)> f_insert_min = [&parser](const boost::python::tuple& args, const boost::python::dict& kw) { return insert_minmaxoneof_<ValueRef::OpType::MINIMUM>(parser, args, kw); };
+    globals["Min"] = boost::python::raw_function(f_insert_min, 3);
```

Comment:
I guess this is refactoring, so maybe this is better for a separate change, but `"Min"` seems overly general and nonspecific here. Maybe store it with the key `"MinOperation"`?

Reply:
Currently I prefer to reproduce FOCS as close as possible so I tries to remain with same tokens,

Reply:
I didn't think the key affected how the parser worked...